instructor_notes/11_git-init-server.md

```diff
@@ -0,0 +1,116 @@
+Note that `usethis::use_git_config()` will not modify (or delete!) other fields besides those provided to the function.
+
+```r
+# Step 1: Create config file with user and email
+usethis::use_git_config(
+    user.name = "Participant Name",
+    user.email = "Participant Email",
+    # Optional setting to ensure the config is global for the user
+    scope = "user",
+    # Optional setting to use `main` as default branch when creating a new repo
+    init.defaultBranch = "main"
+)
+# Step 2: Set up credential helper to be able to temporarily store PAT on the server
+# Note that this could be combined with step 1 as an additional argument
+usethis::use_git_config(
+    credential.helper = "cache --timeout=43200" # 12 hours
+)
+```
+
+### Add your Personal Access Token
+
+* Instruct participants to navigate to their GitHub.com account's Developer settings and create a classic token with the `repo` scope that lasts for 7 days.
+  * Once the token is created, copy/paste it and save in a (local) _secure location_.
+  Emphasize that if they have a password manager, they should use it!
+* In the R Console in the RStudio Server, use the `gitcreds` package to store this information:
+  * Run `gitcreds::gitcreds_set()` and paste the PAT into Console.
+  * Then, run `gitcreds::gitcreds_get()` to ensure a PAT has been set.
+* Explain that they will have to repeat this step every 12 hours to be able to continue communicating with GitHub because of how we set up caching on RStudio Server.
+
+## Part 2: Create a git repository
+
+* In the Terminal, navigate to the `RNA-Seq` workshop directory, which participants are going to turn into a git repository and create a repository:
+
+```sh
+# Navigate to the RNA-seq directory
+cd ~/training-modules/RNA-seq
+
+# Create a git repository
+git init
+```
+
+* Before adding and committing, take a moment to look at the `.gitignore` file present in this directory to see what we expect to be added vs ignored.
+
+* Run the following code to add, commit, and push the new repo to GitHub:
+```sh
+# Add all (non-ignored) files to the directory
+# Explain that we generally like to add files one at a time, but this strategy is appropriate for
+#  the context of initializing a repo, especially given the presence of the .gitignore file!
+git add .
+
+# Throw in a git status for good measure (run as often as desired while using git!)
+git status
+
+# Make a commit
+git commit -m "First commit for RNA-seq workshop materials repo"
+```
+
+* Next, we need a corresponding remote repository to push to.
+For this, instruct participants to create a new repo on GitHub.com
+    * _Do not_ initialize the repository with any files (no `README`, `.gitignore`, or `LICENSE`)
+
+* Copy/paste the `https` URL (`{REMOTE-URL}`) and use for the following code back in R:
+```sh
+# Set the remote repo address
+git remote add origin {REMOTE-URL}
+
+# push to main!
+# now, pushing will go to this URL by default
+git push -u origin main
```

Comment:
Yeah....let me actually run through this on a fresh account with no creds yet and confirm all around.

Reply:
> What happens if the branch does not exist remotely? I haven't done this manually in so long that I forgot!

So we have two choices here which may not be mutually exclusive:
- Generally speaking, `-u origin <branch>` is needed the first time you push to a branch that doesn't yet exist on the remote. the `-u` sets/creates it as upstream. Then, you can just do `git push`.
- You can also add (as I did a long time ago) this to `.gitconfig`. This will automatically create a remote branch when pushing, meaning you only ever need `git push`
  ```
  [push]
    autoSetupRemote = true
  ```

Reply:
I added more comments for `-u origin main`, and included at the end tips on how to work with branches moving forward.